src/test/java/com/xfers/model/example/application/SampleLoan.java

```diff
@@ -57,20 +64,40 @@ public static void main(String[] args) {
         String bankAccountId = exampleAddBankAccount(userApiToken);
         List<BankAccount> bankAccounts = exampleListBankAccounts(userApiToken);
 
-        // Step 2, create a loan. There's a need to check whether the loan can be approved 
-        // first by the bank after the loan creation.
-        // after the loan has been approved, disbursement can be made and then finally
-        // we have to notify the bank that the loan has been disbursed. 
+        // If bankAccountId is null, then it means bank account creation failed.
+        // If the bank account is not created, the disbursement processes can't be done.
+        // If the disbursement process can't be done, there is no point in creating loan.
+        if (null == bankAccountId) {
```

Comment:
same as above, how do i know if the creation has failed? is it better to have a method to check if successful & the reason for failure?

Reply:
All the reason is already present in the respective functions. For example, in this case, the reasons is in exampleCreateBankAccount. Also, I need your help in this reason. I don't know all the reason how it can fail, and what should the user do if fail.

Reply:
failure reason will typically be presented in the response as we typically reply with `{"error":"failure reason"}`, status 400

Reply:
as i recall from some stackoverflow page

400 = you (merchant data) fugged up
500 = i fugged up

Reply:
don't know if we adhere to that religiously